streaming/src/test/scala/org/apache/spark/streaming/StreamingContextSuite.scala

```diff
@@ -297,6 +296,23 @@ class StreamingContextSuite extends SparkFunSuite with BeforeAndAfter with Timeo
     Thread.sleep(100)
   }
 
+  test("de-register codahale metrics on stop()") {
```

Comment:
Can you expand the test to cover both registering and deregistering the metrics. so check after start as well as after stop.

Also, rather than count, a more robust check is to find the StreamingSource in MetricSystem.sources. You will have to access private field StreamingContext.streamingSource and MetricSystem.sources.